tck/src/main/java/org/eclipse/microprofile/concurrency/tck/ThreadContextTest.java

```diff
@@ -580,4 +582,161 @@ public void thirdPartyContextProvidersAreIncludedInThreadContext() {
             Thread.currentThread().setPriority(originalPriority);
         }
     }
+    
+     /**
+     * Verify the MicroProfile Concurrency implementation of propagate(), cleared(), and unchanged()
+     * for ThreadContext.Builder.
+     */
+    @Test
+    public void contextControlsForThreadContextBuilder() throws InterruptedException, ExecutionException, TimeoutException {
+    	ThreadContext allContext = ThreadContext.builder()
```

Comment:
The name `allContext` is a bit misleading.  Intuitively, I would expect something named with all to propagate ThreadContext.ALL_REMAINING, clear nothing, and leave nothing unchanged.  A better name might be bufferContext, referring to what it does propagate.